src/backend/distributed/shared_library_init.c

```diff
@@ -2519,6 +2519,17 @@ CitusAuthHook(Port *port, int status)
 		InitializeBackendData();
 		SetBackendDataDistributedCommandOriginator(true);
 	}
+	else
+	{
+		/*
+		 * We set the global PID in the backend data here already to be able to
+		 * do blocked process detection on connections that are opened over a
+		 * replication connection. A replication connection backend will never
+		 * call StartupCitusBackend, which normally sets up the global PID.
```

Comment:
do we need the call in StartupCitusBackend, or can we always do it here?

Reply:
We cannot generate a global PID here, because that requires reading from catalogs. Extracting from the application name is fine though.